### DISCUSSION TITLE
Request to login/oauth/access_token returns error

### DISCUSSION THREAD

**Discussion by likern** [community/community#24602] [Apps, API and Webhooks]
<p>Hello! I tried virtually everything and it’s not working.</p>
<p>I can successfully get <strong>code </strong> and <strong>state.</strong></p>
<p>But, when I try to exchange them to access token</p>
<pre><code>POST https://github.com/login/oauth/access_token
</code></pre>
<p>request fails with no error message and <strong>406 status code</strong>.</p>
<pre><code>likern@likern:~/Breaffy/old-telegram-bot$ curl -v -H "Accept: application/vnd.github.machine-man-preview+json" -X POST "https://github.com/login/oauth/access_token?client_id=Iv1.daa70164e608e4a0&amp;client_secret=0a93af098d3d310ceee6b023b99ab1921b0b1076&amp;code=25734a3fe561f0ea6752&amp;state=1d257fb4-448a-4208-b785-76ff7b8f8346"
* Trying 140.82.118.4...
* TCP_NODELAY set
* Connected to github.com (140.82.118.4) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
* CAfile: /etc/ssl/certs/ca-certificates.crt
CApath: /etc/ssl/certs
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server accepted to use http/1.1
* Server certificate:
* subject: businessCategory=Private Organization; jurisdictionC=US; jurisdictionST=Delaware; serialNumber=5157550; C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=github.com
* start date: May 8 00:00:00 2018 GMT
* expire date: Jun 3 12:00:00 2020 GMT
* subjectAltName: host "github.com" matched cert's "github.com"
* issuer: C=US; O=DigiCert Inc; OU=www.digicert.com; CN=DigiCert SHA2 Extended Validation Server CA
* SSL certificate verify ok.
&gt; POST /login/oauth/access_token?client_id=Iv1.daa70164e608e4a0&amp;client_secret=0a93af098d3d310ceee6b023b99ab1921b0b1076&amp;code=25734a3fe561f0ea6752&amp;state=1d257fb4-448a-4208-b785-76ff7b8f8346 HTTP/1.1
&gt; Host: github.com
&gt; User-Agent: curl/7.58.0
&gt; Accept: application/vnd.github.machine-man-preview+json
&gt;
&lt; HTTP/1.1 406 Not Acceptable
&lt; Date: Wed, 24 Apr 2019 20:40:41 GMT
&lt; Content-Type: text/html
&lt; Transfer-Encoding: chunked
&lt; Server: GitHub.com
&lt; Status: 406 Not Acceptable
&lt; Vary: X-PJAX
&lt; Cache-Control: no-cache
&lt; Set-Cookie: has_recent_activity=1; path=/; expires=Wed, 24 Apr 2019 21:40:41 -0000
&lt; Set-Cookie: ignored_unsupported_browser_notice=false; path=/
&lt; X-Request-Id: e39c1dfb-531d-4244-83c8-35398dac0197
&lt; Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
&lt; X-Frame-Options: deny
&lt; X-Content-Type-Options: nosniff
&lt; X-XSS-Protection: 1; mode=block
&lt; Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
&lt; Expect-CT: max-age=2592000, report-uri="https://api.github.com/_private/browser/errors"
&lt; Content-Security-Policy: default-src 'none'; base-uri 'self'; block-all-mixed-content; connect-src 'self' uploads.github.com www.githubstatus.com collector.githubapp.com api.github.com www.google-analytics.com github-cloud.s3.amazonaws.com github-production-repository-file-5c1aeb.s3.amazonaws.com github-production-upload-manifest-file-7fdce7.s3.amazonaws.com github-production-user-asset-6210df.s3.amazonaws.com wss://live.github.com; font-src github.githubassets.com; form-action 'self' github.com gist.github.com; frame-ancestors 'none'; frame-src render.githubusercontent.com; img-src 'self' data: github.githubassets.com identicons.github.com collector.githubapp.com github-cloud.s3.amazonaws.com *.githubusercontent.com; manifest-src 'self'; media-src 'none'; script-src github.githubassets.com; style-src 'unsafe-inline' github.githubassets.com
&lt; X-GitHub-Request-Id: D112:46021:117A699:1A4BB95:5CC0C9C9
&lt;
* Connection #0 to host github.com left intact
</code></pre>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/likern">@likern</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<p>I search for some solution and i find this where explain the error code: <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406" rel="nofollow noopener">developer.mozilla.org/Status/406</a>.</p>
<p>It seems that you are not send the right list of acceptable values defined in the request’s proactive content negotiation headers.</p>
<p>Did you follow this guide? -&gt; <a href="https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/" rel="nofollow noopener">developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/</a></p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>

**Comment by likern** [comment]
<p>Thank you so much! You really helped me :wink:</p>
<p>I added Accept: ‘application/json’ and this starts working =)</p>

**Comment by klichukb** [comment]
<p>Has anyone contacted support about this?<br>
<a href="https://developer.github.com/apps/building-github-apps/identifying-and-authorizing-users-for-github-apps/" rel="nofollow noopener">https://developer.github.com/apps/building-github-apps/identifying-and-authorizing-users-for-github-apps/</a><br>
Why do they have a warning and a “preview” header requirement above the “/login/oauth/access_token” endpoint?</p>

**Comment by kamilinalnurd** [comment]
<p>Hi guys, i cant deal with this refreshing access_token with using refresh token.</p>
<p>curl --location --request POST ‘<a href="https://github.com/login/oauth/access_token" rel="noopener nofollow ugc">https://github.com/login/oauth/access_token</a>’ <br>
–header ‘Accept: application/json’ <br>
–header ‘Content-Type: application/json’ <br>
–data-raw ‘{<br>
“refresh_token”: YOUR_REFRESH_TOKEN,<br>
“grant_type”:“refresh_token”,<br>
“client_id”: YOUR_CLIENT_ID,<br>
“client_secret”:YOUR_SECRET<br>
}’</p>
<p>it returns 400 if content is json if content type is form data or x-www-form-urlencoded service returns this<br>
404 status<br>
{<br>
“error”: “Not Found”<br>
}<br>
For any help appreciate!</p>

**Reply by wmvm0** [reply]
me too, request https://github.com/login/oauth/access_token 404
